test/unit/calculators/property_fire_safety_payment_calculator_test.rb

```diff
@@ -0,0 +1,152 @@
+        @calculator.year_of_purchase = 2001
+        @calculator.value_of_property = "100000.99"
+        assert_equal @calculator.uprated_value_of_property, 1_000_010
+      end
+    end
+
+    context "#fully_protected_from_costs?" do
+      should "be true if uprated value is under outer London limit and not living in London" do
+        @calculator.stubs(:uprated_value_of_property).returns(PropertyFireSafetyPaymentCalculator::OUTER_LONDON_VALUATION_LIMIT)
+        @calculator.live_in_london = "no"
+        assert @calculator.fully_protected_from_costs?
+      end
+
+      should "be true if uprated value is under inner London lower limit and living in London" do
+        @calculator.stubs(:uprated_value_of_property).returns(PropertyFireSafetyPaymentCalculator::INNER_LONDON_VALUATION_LIMIT)
+        @calculator.live_in_london = "yes"
+        assert @calculator.fully_protected_from_costs?
+      end
+
+      should "be false if uprated value is over 175k and not living in London" do
+        @calculator.stubs(:uprated_value_of_property).returns(PropertyFireSafetyPaymentCalculator::OUTER_LONDON_VALUATION_LIMIT + 1)
+        @calculator.live_in_london = "no"
+        assert_not @calculator.fully_protected_from_costs?
+      end
+
+      should "be true if uprated value is over 325k and living in London" do
+        @calculator.stubs(:uprated_value_of_property).returns(PropertyFireSafetyPaymentCalculator::INNER_LONDON_VALUATION_LIMIT + 1)
+        @calculator.live_in_london = "yes"
+        assert_not @calculator.fully_protected_from_costs?
+      end
+    end
+
+    context "#leaseholder_costs" do
+      context "not living in London with value between outer London limit and one million" do
+        setup do
+          @calculator.stubs(:uprated_value_of_property).returns(PropertyFireSafetyPaymentCalculator::OUTER_LONDON_VALUATION_LIMIT)
+          @calculator.live_in_london = "no"
+        end
+
+        should "return ten thousand if not shared ownership" do
+          @calculator.shared_ownership = "no"
+          assert_equal @calculator.leaseholder_costs, 10_000
+        end
+
+        should "return pecentage owned multipled by ten thousand if shared ownership" do
+          @calculator.shared_ownership = "yes"
+          @calculator.percentage_owned = 0.50
+          assert_equal @calculator.leaseholder_costs, 5_000
+        end
+      end
+
+      context "living in London with value between inner London limit and one million" do
+        setup do
+          @calculator.stubs(:uprated_value_of_property).returns(PropertyFireSafetyPaymentCalculator::INNER_LONDON_VALUATION_LIMIT)
+          @calculator.live_in_london = "yes"
+        end
+
+        should "return fifteen thousand if not shared ownership" do
+          @calculator.shared_ownership = "no"
+          assert_equal @calculator.leaseholder_costs, 15_000
+        end
+
+        should "return pecentage owned multipled by fifteen thousand if shared ownership" do
```

Comment:
Tiny note: here and in the other tests percentage has a typo

Reply:
good spot 👍